### ISSUE TITLE
Angular Integration

### ISSUE THREAD

**Issue by trimbleo** [trimble-oss/modus-web-components#406]
### Description
Use StencilJS framework integration functionality to generate Angular wrappers and publish them to NPM.

[Stencil Angular Framework Integration](https://stenciljs.com/docs/angular)

### Acceptance Criteria

- [ ] Angular wrappers can be automatically generated
- [ ] Angular wrappers can be automatically published as part of the existing build pipeline
~~Generated Angular wrappers should **not** be stored in version control~~

**Comment by trimbleo** [comment]
@ryan-henness-trimble

## TLDR

In the process of building the Angular library from the stencil web components, I have found all of the manual steps that need to be done to build the library. We originally agreed that the best approach would be to **generate everything on the fly** in the github action. I haven't actually tried this yet but I have ran into a few obvious hurdles with the **generate everything on the fly** approach.

I think it would be much better use of our time to just accept that we might need to create an angular-workspace inside source control that will store the generated components.

## Details

Here are the steps that need to be performed to build an angular library and publish it to npm (Not in perfect order):

**Bold items would be difficult / time consuming to implement using the "generate on the fly" approach**

1. Install the angularOutputTarget dev dependency
`> npm install @stencil/angular-output-target --save-dev`

2. Update the stencil.config.ts to include:
```
angularOutputTarget({
      componentCorePackage: "@trimble-oss/modus-web-components",
      directivesProxyFile: "./angular-workspace/projects/trimble-oss/modus-angular-components/src/lib/stencil-generated/components.ts",
      directivesArrayFile: "./angular-workspace/projects/trimble-oss/modus-angular-components/src/lib/stencil-generated/index.ts",
    })
```

3. Create an angular workspace
`> npx -p @angular/cli ng new angular-workspace --no-create-application`

4.  Create the modus-angular-components library (run from inside the angular workspace)
`> npx -p @angular/cli ng generate library @trimble-oss/modus-angular-components`

**5. Delete the placeholder components and services from the angular library**

6. Build the stencil project (from the stencil project root directory)
`> npm run build`

**7. Import and Export all components into the angular module**

**8. Update tsconfig to disable lib checks. This is required to allow us to extend the stencil HTML elements in Angular**

9. Link the stencil root directory to the angular directory using `npm link`

**10. Bump the angular library version number**

11. Build the angular library:
`> npx ng run @trimble-oss/modus-angular-components:build:production`

12. Publish the angular library:
`> npm publish`

**Comment by ryan-henness-trimble** [comment]
I started wondering yesterday after our talk if this might be too much of a time burner. Sounds like it might be a good idea to just include them in our source control (like you said), as long as we find a good way to let developers know that they are not the source of truth and shouldn't be modified directly

**Comment by trimbleo** [comment]
Agreed.
It can be part of the README.md or CONTRIBUTING.md
I will raise a PR in the coming days for the first part of the angular stuff.

I tried to publish the library to npm yesterday but it said I didn't have paid access to the repo or something. Any ideas there?

**Comment by ryan-henness-trimble** [comment]
Ohh this might be easiest to test out publishing on the public npm registry, since I'm not sure which permissions you'll need to publish on the GitHub registry. I can get you the auth token needed for that
